### ISSUE TITLE
txt2img2img does not work.

### ISSUE THREAD

**Issue by praul** [ThereforeGames/unprompted#53]
Hi there, unfortunately there seems to be a problem with txt2img2img, or setting parameters for img2img:
This is the supplied examples/txt2img2img file:

```Error running postprocess: C:\Apps\stable-diffusion-webui\extensions\unprompted\scripts\unprompted.py<00:00,  3.26it/s]
Traceback (most recent call last):
  File "C:\Apps\stable-diffusion-webui\extensions\unprompted\lib_unprompted\shortcodes.py", line 117, in render
    return str(self.handler(self.token.keyword, self.pargs, self.kwargs, context))
  File "C:\Apps\stable-diffusion-webui\extensions\unprompted\lib_unprompted\shared.py", line 61, in handler
    return(self.shortcode_objects[f"{keyword}"].run_atomic(pargs, kwargs, context))
  File "C:\Apps\stable-diffusion-webui\extensions\unprompted/shortcodes\stable_diffusion\img2img.py", line 20, in run_atomic
    img2img_result = modules.img2img.img2img(
  File "C:\Apps\stable-diffusion-webui\modules\img2img.py", line 96, in img2img
    assert 0. <= denoising_strength <= 1., 'can only work with strength in [0.0, 1.0]'
TypeError: '<=' not supported between instances of 'float' and 'NoneType'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Apps\stable-diffusion-webui\modules\scripts.py", line 375, in postprocess
    script.postprocess(p, processed, *script_args)
  File "C:\Apps\stable-diffusion-webui\extensions\unprompted\scripts\unprompted.py", line 398, in postprocess
    Unprompted.shortcode_objects[i].after(p,processed)
  File "C:\Apps\stable-diffusion-webui\extensions\unprompted/shortcodes\basic\after.py", line 26, in after
    self.Unprompted.process_string(self.Unprompted.parse_alt_tags(content,"after"))
  File "C:\Apps\stable-diffusion-webui\extensions\unprompted\lib_unprompted\shared.py", line 146, in parse_alt_tags
    return(self.shortcode_parser.parse(string,context))
  File "C:\Apps\stable-diffusion-webui\extensions\unprompted\lib_unprompted\shortcodes.py", line 219, in parse
    return stack.pop().render(context)
  File "C:\Apps\stable-diffusion-webui\extensions\unprompted\lib_unprompted\shortcodes.py", line 58, in render
    return ''.join(child.render(context) for child in self.children)
  File "C:\Apps\stable-diffusion-webui\extensions\unprompted\lib_unprompted\shortcodes.py", line 58, in <genexpr>
    return ''.join(child.render(context) for child in self.children)
  File "C:\Apps\stable-diffusion-webui\extensions\unprompted\lib_unprompted\shortcodes.py", line 121, in render
    raise ShortcodeRenderingError(msg) from ex
lib_unprompted.shortcodes.ShortcodeRenderingError: An exception was raised while rendering the 'img2img' shortcode in line 1.```

**Comment by ThereforeGames** [comment]
Hi @praul,

Thanks for bringing this to my attention - it looks like the img2img syntax changed in recent versions of the WebUI.

I have fixed this internally by updating the `img2img_result` block in `unprompted/shortcodes/stable_diffusion/img2img.py` to the following:

```
		img2img_result = modules.img2img.img2img(
			"unprompted_img2img",
			self.Unprompted.shortcode_user_vars["img2img_mode"] if "img2img_mode" in self.Unprompted.shortcode_user_vars else 0, #p.mode
			self.Unprompted.shortcode_user_vars["prompt"],
			self.Unprompted.shortcode_user_vars["negative_prompt"],
			"None", # prompt_styles
			self.Unprompted.shortcode_user_vars["init_images"][len(self.Unprompted.after_processed.images) - 1],
			None, # sketch
			None, # p.init_img_with_mask
			None, # inpaint_color_sketch
			None, # inpaint_color_sketch_orig
			self.Unprompted.shortcode_user_vars["init_images"][len(self.Unprompted.after_processed.images) - 1], # p.init_img_inpaint
			init_mask, # p.init_mask_inpaint
			self.Unprompted.shortcode_user_vars["steps"],
			sampler_index,
			self.Unprompted.shortcode_user_vars["mask_blur"] if "mask_blur" in self.Unprompted.shortcode_user_vars else 0, # p.mask_blur
			0.0, #p.mask_alpha
			0, # p.inpainting_fill
			self.Unprompted.shortcode_user_vars["restore_faces"],
			self.Unprompted.shortcode_user_vars["tiling"],
			self.Unprompted.shortcode_user_vars["n_iter"] if "n_iter" in self.Unprompted.shortcode_user_vars else 1, #p.n_iter
			self.Unprompted.shortcode_user_vars["batch_size"] if "batch_size" in self.Unprompted.shortcode_user_vars else 1, #p.batch_size
			self.Unprompted.shortcode_user_vars["cfg_scale"],
			self.Unprompted.shortcode_user_vars["denoising_strength"] if self.Unprompted.shortcode_user_vars["denoising_strength"] is not None else 0.5,
			self.Unprompted.shortcode_user_vars["seed"],
			self.Unprompted.shortcode_user_vars["subseed"],
			self.Unprompted.shortcode_user_vars["subseed_strength"],
			self.Unprompted.shortcode_user_vars["seed_resize_from_h"],
			self.Unprompted.shortcode_user_vars["seed_resize_from_w"],
			0, # seed_enable_extras
			self.Unprompted.shortcode_user_vars["height"],
			self.Unprompted.shortcode_user_vars["width"],
			self.Unprompted.shortcode_user_vars["resize_mode"] if "resize_mode" in self.Unprompted.shortcode_user_vars else 0,
			self.Unprompted.shortcode_user_vars["inpaint_full_res"] if "inpaint_full_res" in self.Unprompted.shortcode_user_vars else True, # p.inpaint_full_res
			self.Unprompted.shortcode_user_vars["inpaint_full_res_padding"] if "inpaint_full_res_pdding" in self.Unprompted.shortcode_user_vars else 0, # p.inpaint_full_res_padding
			0, # p.inpainting_mask_invert
			"", #p.batch_input_directory
			"", #p.batch_output_directory
			0, # this is the *args tuple and I believe 0 indicates we are not using an extra script in img2img
		)
```

Hope that helps.

I will leave this issue open until the fix goes live.

**Comment by ThereforeGames** [comment]
Resolved in v6.0.0.

**Comment by praul** [comment]
Hi and thanks for quick solving approach. However, using the supplied example now, does not seem to do img2img but rather create two different pictures. Console also errors:

```
Error running process: C:\Apps\stable-diffusion-webui\extensions\unprompted\scripts\unprompted.py0:05<00:00,  3.32it/s]
Traceback (most recent call last):
  File "C:\Apps\stable-diffusion-webui\modules\scripts.py", line 357, in process
    script.process(p, *script_args)
TypeError: Scripts.process() missing 2 required positional arguments: 'is_enabled' and 'unprompted_seed'

```

![Screenshot 2023-01-26 083719](https://user-images.githubusercontent.com/38807639/214781244-11d08776-966d-4d7d-aad4-ea2c9a1021a3.png)


Additionally, when launching console warns
```
C:\Apps\stable-diffusion-webui\extensions\unprompted\scripts\unprompted.py:73: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if gr_obj.value is None or gr_obj.value is "": continue # Skip empty fields
```

I don't know if this is related. Thanks for your work :)
